model/src/presignature.rs

```diff
@@ -0,0 +1,45 @@
+//! Contains the Presignature type as described by the specification with serialization as described by the openapi documentation.
+
+use crate::order::OrderUid;
+use primitive_types::H160;
+use serde::{Deserialize, Serialize};
+
+#[derive(Eq, PartialEq, Clone, Debug, Default, Deserialize, Serialize, Hash)]
+#[serde(rename_all = "camelCase")]
+pub struct PreSignature {
+    pub block_number: u64,
+    pub log_index: u64,
+    pub owner: H160,
+    pub order_uid: OrderUid,
+    pub signed: bool,
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use serde_json::json;
+
+    #[test]
+    fn deserialization_and_back() {
+        let value = json!(
+        {
+            "blockNumber": 1337u64,
+            "logIndex": 42u64,
+            "owner": "0x0000000000000000000000000000000000000042",
```

Comment:
Offtopic: H160 serializes like this? Then why did we need H160Hexadecmial? Did this change recently?

Reply:
This was already the case since the first of April, [here](https://github.com/gnosis/gp-v2-services/blob/main/model/src/trade.rs) is the code on which this section is based.